logic/src/commonMain/kotlin/com/wire/kalium/logic/GlobalKaliumScope.kt

```diff
@@ -105,28 +105,30 @@ class GlobalKaliumScope internal constructor(
         )
     }
 
-    internal val serverConfigRepository: ServerConfigRepository
-        get() = ServerConfigDataSource(
+    internal val serverConfigRepository: ServerConfigRepository by lazy {
+        ServerConfigDataSource(
             unboundNetworkContainer.serverConfigApi,
             globalDatabase.serverConfigurationDAO,
             unboundNetworkContainer.remoteVersion,
             kaliumConfigs.developmentApiEnabled
         )
+    }
 
-    val sessionRepository: SessionRepository
-        get() = SessionDataSource(
+    val sessionRepository: SessionRepository by lazy {
```

Comment:
this will make all of those Repository as singelton, and they are already created on demand so not sure if this will change anything

Reply:
by lazy i would assume something like

```
class MyScope(
    val conversationRepo: () -> ConversationRepo 
) {

val conversationRepository by lazy { conversationRepo() }

}
```

this will delay creating the repos inside the scope and still use the same instance inside the scope

Reply:
or we can delegate creating those repos for auser to another class and inject that into the scopes 

```
class UserRepoProvider(
    val userStorage: UserStorage
    val userNetworkContainer: AuthenticatedNetworkContainer
)  {

    fun provideConversationRepository(): ConversationRepository

}


class TeamScope(
    val repoProvider: UserRepoProvider
) {

// create usse cases
}